dipy/segment/metricspeed.pyx

```diff
@@ -44,12 +44,12 @@ cdef class Metric(object):
         def __get__(Metric self):
             return bool(self.is_order_invariant)
 
-    cdef int c_are_compatible(Metric self, Shape shape1, Shape shape2) nogil:
+    cdef int c_are_compatible(Metric self, Shape shape1, Shape shape2) nogil except -1:
         """ Cython version of `Metric.are_compatible`. """
         with gil:
             return self.are_compatible(shape2tuple(shape1), shape2tuple(shape2))
 
-    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil:
+    cdef double c_dist(Metric self, Data2D features1, Data2D features2) nogil except -1.0:
```

Comment:
I'm not sure. I think it could be a int but will be converted in double anyway.
http://docs.cython.org/src/userguide/language_basics.html#error-return-values